src/sd.jl

```diff
@@ -679,6 +679,23 @@ function build_constraint(
     )
 end
 
+function build_constraint(
+    _error::Function,
+    Q::AbstractMatrix{<:AbstractJuMPScalar},
+    cone::HermitianPSDCone,
+)
+    if !LinearAlgebra.ishermitian(Q)
+        _error(
+            "Unable to add matrix in HermitianPSDCone because the matrix is " *
+            "not Hermitian. If this error was unexpected, check the matrix " *
+            "for errors, or wrap the matrix `H` in " *
```

Comment:
It's unclear what you mean by "check the matrix for errors"

Reply:
"because the matrix is of type `LinearAlgebra.Hermitian`. Even if the matrix is `H`, this need to be made explicit by wrapping it as `LinearAlgebra.Hermtian(H)`."

Reply:
Note that this method does check for `LinearAlgebra.ishermitian`

Reply:
Ah, didn't catch that, I would just error all the time. This is a remake of https://github.com/jump-dev/JuMP.jl/pull/3264

Reply:
Ah. I thought we'd done it before, but I couldn't find.

Remind me again why? Just because it's inconsistent if you query `dual(con)` that you get a `Hermitian` matrix back when you passed a non-Hermitian square matrix? 

It really seems weird that the case in https://github.com/jump-dev/JuMP.jl/issues/3367 doesn't work.

Reply:
If we want it to work, we can add the square cone in MOI. The issue with this PR is that when we do that then the type returned by `dual` will change in JuMP so that will be breaking.
So it's best to have an issue here and implement the square cone in MOI if there is a need for it.

Reply:
Updated to just throw the error